src/runtime-rs/crates/resource/src/network/policy_rule.rs

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) 2019-2022 Alibaba Cloud
+// Copyright (c) 2019-2022 Ant Group
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use agent::{IPFamily, Rule};
+use anyhow::{Context, Result};
+use futures::stream::TryStreamExt;
+use netlink_packet_route::rule::RuleMessage;
+use std::net::IpAddr;
+
+pub(crate) struct NetworkRule {
+    pub(crate) rules: Vec<Rule>,
+}
+
+impl NetworkRule {
+    pub async fn new(handle: &rtnetlink::Handle) -> Result<NetworkRule> {
+        let rule_list = handle_rules(handle).await.context("handle rules")?;
+
+        Ok(NetworkRule { rules: rule_list })
+    }
+}
+
+fn octets_to_addr(octets: &[u8], prefix_len: u8) -> Option<(IpAddr, u8)> {
+    if octets.len() == 4 {
+        let mut ary: [u8; 4] = Default::default();
+        ary.copy_from_slice(octets);
+        Some((IpAddr::from(ary), prefix_len))
+    } else if octets.len() == 16 {
+        let mut ary: [u8; 16] = Default::default();
+        ary.copy_from_slice(octets);
+        Some((IpAddr::from(ary), prefix_len))
+    } else {
+        None
+    }
```

Comment:
Please optimize the code:
- match {} may be a better way.
- and some code are dup.

Reply:
Thx，fixed